sor/src/main/java/com/bazaarvoice/emodb/sor/core/DistributedCompactor.java

```diff
@@ -158,6 +159,11 @@ protected Expanded doExpand(Record record, long fullConsistencyTimestamp, long c
             // Re-initialize the resolver with the new compaction state.
             resolver = new DefaultResolver(intrinsics, compaction);
         }
+
+        // consider deleteDeltasTimestamp and not include those Ids.
```

Comment:
I'd considered recommending an alternative approach with checking for `deleteDeltasTimestamp` at insertion point into `keysToDelete` rather than adding to and then removing from the list.  However, I think what you've done here is cleaner and gives a better separation of the two concerns.  My only comment is to better comment what is going on here, since to me it's non-obvious why the values inserted into `keysToDelete` are suddenly no good.